### ISSUE TITLE
Finetune on SA-1B dataset for all instance segmentation

### ISSUE THREAD

**Issue by amish1706** [script-Yang/segdino#12]
Wanted to know if this training method can be used to finetune the head for all instance segmentation task (class agnostic) using the SA-1B dataset released by Meta for SAM model.
What changes would be required in the architecture of segmentation head for this task?

**Comment by script-Yang** [comment]
My understanding of your question is that you want to fine-tune the segmentation head for class-agnostic instance segmentation (for example, using Meta’s SA-1B dataset), enabling the model to distinguish different object instances without relying on semantic class labels. Since the original DPT head performs only class-aware semantic segmentation, it cannot separate multiple instances of the same type. To address this, the head can be modified to output three branches: a foreground mask, an instance center heatmap, and a per-pixel offset vector (dx, dy), allowing instance grouping based on geometric relationships rather than category labels. You can refer to the following implementation; however, I have not tested this approach on large-scale datasets, so its ability to scale up remains uncertain.

```py
# --- DPTHead modification ---
self.mask_head   = nn.Conv2d(features * 4, 1, kernel_size=1, stride=1, padding=0)  # foreground logits
self.center_head = nn.Conv2d(features * 4, 1, kernel_size=1, stride=1, padding=0)  # center heatmap logits
self.offset_head = nn.Conv2d(features * 4, 2, kernel_size=1, stride=1, padding=0)  # (dx, dy)

# --- forward() return ---
fused = torch.cat([layer_1_rn, layer_2_up, layer_3_up, layer_4_up], dim=1)
mask_logit   = self.mask_head(fused)
center_logit = self.center_head(fused)
offset       = self.offset_head(fused)
return {"mask_logit": mask_logit, "center_logit": center_logit, "offset": offset}
```
